streaming/src/main/scala/org/apache/spark/streaming/api/java/JavaDStreamLike.scala

```diff
@@ -284,50 +264,6 @@ trait JavaDStreamLike[T, This <: JavaDStreamLike[T, This, R], R <: JavaRDDLike[T
   /**
    * Apply a function to each RDD in this DStream. This is an output operator, so
    * 'this' DStream will be registered as an output stream and therefore materialized.
-   *
-   * @deprecated  As of release 0.9.0, replaced by foreachRDD
-   */
-  @deprecated("Use foreachRDD", "0.9.0")
-  def foreach(foreachFunc: JFunction[R, Void]) {
-    foreachRDD(foreachFunc)
-  }
-
-  /**
-   * Apply a function to each RDD in this DStream. This is an output operator, so
-   * 'this' DStream will be registered as an output stream and therefore materialized.
-   *
-   * @deprecated  As of release 0.9.0, replaced by foreachRDD
-   */
-  @deprecated("Use foreachRDD", "0.9.0")
-  def foreach(foreachFunc: JFunction2[R, Time, Void]) {
-    foreachRDD(foreachFunc)
-  }
-
-  /**
-   * Apply a function to each RDD in this DStream. This is an output operator, so
-   * 'this' DStream will be registered as an output stream and therefore materialized.
-   *
-   * @deprecated  As of release 1.6.0, replaced by foreachRDD(JVoidFunction)
-   */
-  @deprecated("Use foreachRDD(foreachFunc: JVoidFunction[R])", "1.6.0")
```

Comment:
I am not sure, but usually deprecated notice stay for a release. This can go after next release either as 1.7.x or 2.0.